lib/compiler_rt/x86_cpu_model.zig

```diff
@@ -0,0 +1,165 @@
+//! Thus file implements x86/x86_64 CPU detection compatible with LLVM/Clang
+//! (i.e. multiversioned functions with attribute `target` or `target_clones`) which
+//! itself is compatible with libgcc multiversioning.
+
+const builtin = @import("builtin");
+const std = @import("std");
+const x86 = std.zig.system.x86;
+
+const common = @import("common.zig");
+
+const Target = std.Target;
+
+comptime {
+    const linkage: std.builtin.GlobalLinkage = common.linkage;
+    const visibility: std.builtin.SymbolVisibility = if (linkage != .internal) .hidden else .default;
+
+    @export(&cpu, .{ .name = "__cpu_model", .linkage = linkage, .visibility = visibility });
+    @export(&cpu_extra_features, .{ .name = "__cpu_features2", .linkage = linkage, .visibility = visibility });
+    @export(&init, .{ .name = "__cpu_indicator_init", .linkage = linkage, .visibility = visibility });
+}
+
+var cpu: Model = .{};
+var cpu_extra_features: [feature_set_size - 1]u32 = [_]u32{0} ** (feature_set_size - 1);
+
+// TODO: Do we need constructor attribute? As far as I can tell the generated LLVM detection code explicitly
+//       calls __cpu_indicator_init before actually using __cpu_model or __cpu_features2 (happens in
+//       clang/lib/CodeGen/CGBuiltin.cpp)
+fn init() callconv(.C) c_int {
+    if (cpu.vendor != .unknown) return 0;
```

Comment:
Since this can be called from multiple threads, this check should be replaced with an atomic flag using unordered loads/stores.

Reply:
Should it be acquire-release instead to ensure ordering with respect to the other fields?